src/wasm.rs

```diff
@@ -30,9 +40,16 @@ impl HttpClient for WasmClient {
         'a: 'async_trait,
         Self: 'async_trait,
     {
+        let config = self.config.clone();
+
         InnerFuture::new(async move {
             let req: fetch::Request = fetch::Request::new(req).await?;
-            let mut res = req.send().await?;
+            let conn = req.send();
+            let mut res = if let Some(timeout) = config.timeout {
+                async_std::future::timeout(timeout, conn).await??
```

Comment:
Hi @Fishrock123, just out of curiosity is this line ok? I think should be one `?`.

Thanks!!

Reply:
Yes, this does compile, it is roughly equivalent to the following:
```rust
let mut res = if let Some(timeout) = config.timeout {
    async_std::future::timeout(timeout, conn).await?
} else {
    conn.await
}?;
```

Basically... timeout has it's own error and then the connection await has it's error... this is like a cheap-seats `Result::flatten()` (which is still unstable).

Reply:
Awesome 🙌, thanks for the explanation!!